test/farm/001-farm.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+#
+# Tests of podman farm commands
+#
+
+load helpers.bash
+
+###############################################################################
+# BEGIN tests
+
+fname="test-farm"
+containerfile="test/farm/Containerfile"
+
+@test "farm - check farm has been created" {
+    run_podman farm ls
+    assert "$status" -eq 0 "get list of farms"
+    assert "$output" =~ $fname
+    assert "$output" =~ "test-node"
+}
+
+@test "farm - build on local only" {
+    iname="test-image-1"
+    empty_farm="empty-farm"
+    # create an empty farm
+    run_podman farm create $empty_farm
+    assert "$status" -eq 0 "create empty farm"
+    run_podman farm --farm $empty_farm build -f $containerfile -t $iname .
+    assert "$status" -eq 0 "create new manifest list $iname"
+    assert "$output" =~ "Local builder ready"
+    # inspect manifest list built and saved in dir
+    run_podman manifest inspect $iname
+    assert "$status" -eq 0 "inspect manifest list $iname"
+    assert "$output" =~ "amd64"
+    # cleanup images on node
+    ssh $ROOTLESS_USER@localhost podman rmi -af
+    assert "$status" -eq 0 "remove built image on remote node"
+}
+
+@test "farm - build on farm node and local" {
+    iname="test-image-2"
+    run_podman farm build -f $containerfile -t $iname .
+    assert "$status" -eq 0 "create new manifest list $iname"
+    assert "$output" =~ "Farm \"$fname\" ready"
+    # inspect manifest list built and saved in dir
+    run_podman manifest inspect $iname
+    assert "$status" -eq 0 "inspect manifest list $iname"
+    assert "$output" =~ "amd64"
+    # cleanup images on node
+    ssh $ROOTLESS_USER@localhost podman rmi -af
+    assert "$status" -eq 0 "remove built image on remote node"
```

Comment:
I know I mentioned all the `assert status` lines, but this one is particularly wrong: `$status` is only meaningful after a `run` command. In this case, if the `ssh` on line 49 fails, that's the end of the test, it fails right there. And if it succeeds, this is a misleading NOP because the `$status` being tested will be that from a long-ago (line 45) `run`.

Reply:
Fixed